### PR TITLE
FlattenJson filter

### PR DESCRIPTION
**Pull request by robbiezhang**
Filter to select a node, split if it's an array
and flatten the schema of the inner json object

Usage:
<filter>
  type filter_flatten
  select record['apps']['app']
</filter>

### CODE REVIEW

**Comment by robbiezhang** [comment]
@Microsoft/omsagent-devs

**Comment by jeffaco** [comment]
@niroyb is on vacation for a week today and all next week, and I dunno enough about Ruby to comment on this. Don't wait for Nick ...

**Comment by robbiezhang** [comment]
@varunkumta @sw47 Nick's OOF this week, can you take a look?

File: test/code/plugins/flattenjson_lib_test.rb [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  def teardown
+  end
+
+  def test_flatten_target_array
+    es = MockEventStream.new
+
+    record = { 'root' => { 'target' => [ {'a' => 'b', 'c' => {'d' => 'e'}, 'f' => [ 1, 2, 3] }, { 'g' => { 'h' => [ {'i' => 4 }, { 'j' => 5 } ] }} ] } }
+    @@flattenjson_lib.select_split_flatten(nil, record, 'record["root"]["target"]', es)
+
+    assert_equal(2, es.length, "Expect 1 record");
```
Comment (sw47) [inline test/code/plugins/flattenjson_lib_test.rb]: minor: 2 records

File: source/code/plugins/flattenjson_lib.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+module OMS
+  require_relative 'omslog'
+
+  class FlattenJson
+    def select_split_flatten(time, record, select, es)
```
Comment (sw47) [inline source/code/plugins/flattenjson_lib.rb]: Some comments on these functions will be nice.

File: source/code/plugins/filter_flatten.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    def start
+      super
+      @flattenjson_lib = OMS::FlattenJson.new
+    end
+
+    def shutdown
+      super
+    end
+
+    def filter_stream(tag, es)
```
Comment (sw47) [inline source/code/plugins/filter_flatten.rb]: Put the usage as a comment here.

**Comment by sw47** [comment]
Signed off

**Pull request merged**